src/common/simulation.ts

```diff
@@ -338,50 +370,77 @@ export default abstract class Simulation {
     body.setLinearVelocity(velocity);
     body.setAwake(true);
   }
-}
-
-function createWorld(map: GameMap): World {
-  const world = new World();
 
-  const fixDef: any = {
-    friction: 0.0,
-    restitution: 0.0,
-  };
+  createWorld(map: GameMap): World {
+    const world = new World();
 
-  const halfSize = 0.5 * constants.TILE_LOGICAL_SIZE;
-
-  for (let y = 0; y < map.height; ++y) {
-    for (let x = 0; x < map.width; ++x) {
-      if (!map.at(x, y).walkable) {
-        const center = new Vec2(
-          x * constants.TILE_LOGICAL_SIZE + halfSize,
-          y * constants.TILE_LOGICAL_SIZE + halfSize,
-        );
-        const shape: any = new Box(halfSize, halfSize, Vec2.zero(), 0.0);
-
-        const body = world.createBody({
-          type: Body.STATIC,
-          position: center,
-          fixedRotation: true,
-          active: true,
-          awake: false,
-        });
+    const fixDef: any = {
+      friction: 0.0,
+      restitution: 0.0,
+    };
 
-        body.createFixture(shape, fixDef);
+    const halfSize = 0.5 * constants.TILE_LOGICAL_SIZE;
+
+    for (let y = 0; y < map.height; ++y) {
+      for (let x = 0; x < map.width; ++x) {
+        if (!map.at(x, y).walkable) {
+          const center = new Vec2(
+            x * constants.TILE_LOGICAL_SIZE + halfSize,
+            y * constants.TILE_LOGICAL_SIZE + halfSize,
+          );
+          const shape: any = new Box(halfSize, halfSize, Vec2.zero(), 0.0);
+
+          const body = world.createBody({
+            type: Body.STATIC,
+            position: center,
+            fixedRotation: true,
+            active: true,
+            awake: false,
+          });
+
+          body.createFixture(shape, fixDef);
+        }
       }
     }
-  }
 
-  // TODO: hantera kollisioner om något speciellt ska hända
-  // world.on('begin-contact', contact => {
-  //   let a = contact.getFixtureA(),
-  //     b = contact.getFixtureB();
+    // TODO: hantera kollisioner om något speciellt ska hända
+    world.on('begin-contact', contact => {
+      const user_data_a = contact
+        .getFixtureA()
+        .getBody()
+        .getUserData() as { id: number };
+      const user_data_b = contact
+        .getFixtureB()
+        .getBody()
+        .getUserData() as { id: number };
+      let player_id: number | undefined = undefined;
+      let enemy_id: number | undefined = undefined;
+      if (this._state.players[user_data_a?.id] !== undefined) {
+        player_id = user_data_a?.id;
+        enemy_id = user_data_b?.id;
+      }
+      if (this._state.players[user_data_b?.id] !== undefined) {
+        player_id = user_data_b?.id;
+        enemy_id = user_data_a?.id;
+      }
+      if (
+        enemy_id !== undefined &&
+        player_id !== undefined &&
+        this._state.enemies[enemy_id] !== undefined
+      ) {
```

Comment:
`if (enemy_id && player_id && this._state.enemies[enemy_id])`
Så kanske allt får plats på en rad och då blir Carl glad